x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/TypeResolutions.java

```diff
@@ -45,6 +46,29 @@ public static TypeResolution isIP(Expression e, String operationName, ParamOrdin
         return isType(e, dt -> dt == IP, operationName, paramOrd, "ip");
     }
 
+    public static TypeResolution isEnum(Expression e, String operationName, ParamOrdinal paramOrd, Set<String> acceptedValues) {
+        TypeResolution resolution = isFoldable(e, operationName, paramOrd);
+        if (resolution.unresolved()) {
+            return resolution;
+        }
+
+        resolution = isString(e, operationName, paramOrd);
+        if (resolution.unresolved()) {
+            return resolution;
+        }
+
+        String value = (String)e.fold();
+        if (acceptedValues.contains(value) == false) {
+            return new TypeResolution(format(null, "{}argument of [{}] must be a string, one of {}, received [{}]",
```

Comment:
The error message is a bit messed up. This is how it looks with actual values in it: `third argument of [PERCENTILE(int, 50, null)] must be a string, one of [tdigest, hdr], received [null]`. How about `third argument of [PERCENTILE(int, 50, null)] must be one of [tdigest, hdr], received [null]`? To me it seems more clear and concise.